src/NClient.CodeGeneration/NClient.CodeGeneration.Abstractions/INClientGenerator.cs

```diff
@@ -0,0 +1,10 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NClient.CodeGeneration.Abstractions
+{
+    public interface INClientGenerator
+    {
+        Task<string?> GenerateAsync(string specification, string @namespace, CancellationToken cancellationToken = default);
```

Comment:
Generation does not have enough settings, for example, enabling/disabling the addition of cancellation tokens, changing collection types in parameters, and so on

Reply:
Do you want to copy all NSwag generation settings?

Reply:
Which are relevant to interfaces

Reply:
It's complicated. Maybe it's not necessary for MVP.
If we want to create wrapper settings, I suggest to add smth like "generation predicate" and it will entail some arch difficulties right now.

Reply:
Okay, let's do it later